packages/@vuepress/core/lib/node/build/index.js

```diff
@@ -90,10 +91,16 @@ module.exports = class Build extends EventEmitter {
 
     // render pages
     logger.wait('Rendering static HTML...')
-
-    const pagePaths = await Promise.all(
-      this.context.pages.map(page => this.renderPage(page))
-    )
+    if (this.maxConcurrency) logger.info(`max concurrency set: ${this.maxConcurrency}`)
+
+    const pagePaths = []
+    while (this.context.pages.length) {
+      const segmentPaths = await Promise.all(
+        this.context.pages.splice(0, this.maxConcurrency || 100000)
```

Comment:
It is not advisable to modify `context.pages` directly

Reply:
changed to slice + for loop